File: src/main/java/io/hangman/service/UserService.java
package io.hangman.service;

import java.util.List;

import io.hangman.dto.UserDto;
import io.hangman.dto.WordDto;
import io.hangman.entity.User;

public interface UserService {
	static int MAX_FAILURES = 6;
	
	UserDto reset(UserDto userDto, WordDto wordDto);
	UserDto checkLetter(Integer userId, String letter);
    UserDto getUserById(Integer userId);
    void saveUser(UserDto userDto);
    List<UserDto> getAllUsers();
    List<User> getAllUsersNaked();
}

File: src/main/resources/static/js/app/UserService.js
'use strict'

angular.module('desafio.services', []).factory('UserService',
		[ "$http", "CONSTANTS", function($http, CONSTANTS) {
			var service = {};
			service.reset = function(userId) {
				var url = CONSTANTS.reset + userId;
				return $http.put(url, null);
			}
			service.checkLetter = function(userId, letter) {
				var url = CONSTANTS.checkLetter + userId + '/' + letter;
				return $http.put(url, null);
			}
			service.getAllUsers = function() {
				return $http.get(CONSTANTS.getAllUsers);
			}
			return service;
		} ]);